R/dp_accessors.R

```diff
@@ -203,6 +203,33 @@ get_women_on_art <- function(tag, tag_data, metadata, agegr_labels) {
   women_on_art
 }
 
+#' Get ART eligibility population data.
+#'
+#' @param tag The name of the tag getting data for.
+#' @param tag_data Tag data to be parsed.
+#' @param metadata Metadata params made available by tag configuration.
+#' @param agegr_labels Age group labels.
+#'
+#' @keywords internal
+get_eligibility_pop_data <- function(tag, tag_data, metadata, proj_years) {
+  if (is.null(metadata$rows) || is.null(metadata$cols)) {
+    stop(sprintfr("Can't get eligibility pop data via tag %s. Configuration is
+                  incomplete. Must specify rows and cols at minimum.
+                  rows are null: %s, cols are null: %s.",
+                  tag, is.null(metadata$rows), is.null(metadata$cols)))
+  }
+
+  data <- tag_data[metadata$rows, metadata$cols]
+  data <- stats::setNames(data, c("description", "pop", "elig", "percent", "year"))
+  data$pop <- c("PW", "TBHIV", "DC", "FSW", "MSM", "IDU", "OTHER")
+  data$elig <- as.logical(as.integer(data$elig))
```

Comment:
how strongly do we feel about the abbreviations in returned data? (here, `pop`, `elig` and `idx`). Especially with the main functions being `get_eligibility_pop_data`

Reply:
So far I've been trying to avoid abbreviations within my own code unless with something which is used very commonly like `pop` for `population`. 

The reason I am using the abbreviations here is that these are the names returned by `first90` meaning that when this is implemented it should be able to be dropped in place of the current `firt90` impl without too much integration headache. Which I think is a good idea as I expect that these properties are being accessed in a few places. I then at a later point want to review this generally with Jeff and move to using more meaningful names.

So see e.g. https://github.com/mrc-ide/specio/blob/i171/R/read_hivproj_param.R#L73 where I am mapping between my naming and the current naming returned by Jeff's code. I'm hoping to fix all this in [another ticket](https://vimc.myjetbrains.com/youtrack/issue/mrc-174).